### PR TITLE
Update macOS build to include app bundle related files

### PR DESCRIPTION
**Pull request by sharvilshah**
This change includes all the control files needed by the app bundle on macOS.

This PR in packaging repo (https://github.com/osquery/osquery-packaging/pull/3) is a prerequisite

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -443,9 +443,17 @@ function(generateInstallDirectives)
     )
 
   elseif(PLATFORM_MACOS)
+
+    if(CMAKE_INSTALL_PREFIX_INITIALIZED_TO_DEFAULT)
+      set(CMAKE_INSTALL_PREFIX "/opt" CACHE PATH "" FORCE)
```
Comment (directionless) [inline CMakeLists.txt:448]: Should probably be `/opt/osquery`

File: tools/deployment/macos_packaging/PkgInfo [hunk 2]
```diff
@@ -0,0 +1 @@
+APPL????
```
Comment (directionless) [inline tools/deployment/macos_packaging/PkgInfo:1]: Is this a binary file? If so, it should get added to the linguist diff suppression in `.gitattributes`

What is it?

File: tools/deployment/osqueryctl [hunk 3]
```diff
@@ -111,11 +111,11 @@ main() {
     EXAMPLE_CONFIG_PATH="/var/osquery/osquery.example.conf"
     PIDFILE="/var/osquery/osquery.pid"
     LOCKFILE="/var/osquery/osquery.lock"
-    EXEC="/usr/local/bin/osqueryd"
-    PLIST_DOMAIN="com.facebook.osqueryd"
+    EXEC="/opt/osquery.app/Contents/MacOS/osqueryd"
```
Comment (directionless) [inline tools/deployment/osqueryctl:114]: Do we intent to have this land in `/opt/osquery.app` ? I was thinking something more like:
* `/opt/osquery/bin/osqueryd`  -- common to all platforms
* `/opt/osquery/???/osquery.app` -- macOS 

Whether `???` is `lib` `share` or nothing at all I don't know

**Comment by theopolis** [comment]
What's the next step for getting this merged?

**Review by directionless — APPROVED** [review]

**Pull request merged**